documentation/src/main/asciidoc/topics/proc_configuring_textbased_encoding.adoc

```diff
@@ -0,0 +1,33 @@
+[id='using_text_based_storage-{context}']
+= Using Text-Based Formats for Data
+You can configure {brandname} to encode data in text-based formats such as
+`text/plain`, `application/json`, or `application/xml`.
+
+.Procedure
+
+. Specify a text-based storage format as the MediaType for keys and values.
+. Optionally specify a character set such as `UTF-8`.
++
+The following example configures {brandname} to store entries with the `text/plain; charset=UTF-8` format:
++
+[source,xml,options="nowrap",subs=attributes+]
+----
+include::config_examples/encoding_text_plain.xml[]
+----
++
+. Configure your clients.
+
+Hot Rod clients can use `org.infinispan.commons.marshall.StringMarshaller`
+to handle plain text, JSON, XML, or any other text-based format.
+
+If you want to use text-based formats and binary data with Hot Rod clients, you
+should use ProtoStream marshalling. ProtoStream can handle plain text and JSON 
+without the need to create Serialization Contexts and register Protobuf
+schemas. However, ProtoStream does not support XML.
```

Comment:
```suggestion
Text-based formats can also be used with the Protostream marshaller in the Hot Rod client
without the need to create Serialization Contexts and register Protobuf
schemas, since it can handle ```String``` and ```byte[]``` types natively
```